bpf/lib/nodeport.h

```diff
@@ -2401,10 +2402,22 @@ nodeport_rev_dnat_ingress_ipv4(struct __ctx_buff *ctx, struct trace_ctx *trace,
 	bool allow_neigh_map = true;
 	bool check_revdnat = true;
 	bool has_l4_header;
+	__u32 *vrf_id __maybe_unused = NULL;
 
 	if (!revalidate_data(ctx, &data, &data_end, &ip4))
 		return DROP_INVALID;
 
+#if defined(ENABLE_SRV6) && defined(IS_BPF_LXC)
+	/* Determine if packet belongs to a VRF before we do NAT.
```

Comment:
drive-by - will this still work once we finish https://github.com/cilium/cilium/issues/24062, and handle RevDNAT in `to-netdev`?

Reply:
It's a bit tricky because we need to lookup the route for the outer IPv6 header after SRv6 encapsulation. Currently, we do encapsulation in tc-ingress and just pass it to the stack so that the stack can route the packet, but if we are in the tc-egress, we somehow need to put the packet back to stack. It is possible (e.g. by redirecting to device ingress), but a bit weird packet flow.

Reply:
Keep in mind that you already need such support today, for any pod that uses L7 network policies. Otherwise the pod's replies are redirected to the L7 proxy, and never hit the RevDNAT hook that you're using here.

Reply:
Ah, indeed. Currently, there's no immediate need for L7 support, but I'll follow up later.

Reply:
fwiw, what you described further up sounds similar to what [this code section](https://github.com/cilium/cilium/blob/e31b50db6a59270f39f72642e7b4276403a15bb1/bpf/lib/nodeport.h#L3064) solves in the RevDNAT @ `to-netdev` case - applying post-RevDNAT logic, and redirecting as needed.

Reply:
Yeah, that sounds like we'd like to achieve. After revNAT & SRv6 encap, we can do `fib_lookup` against outer IPv6 destination and redirect the packet.